packages/dart/noports_core/lib/src/srv/srv_impl.dart

```diff
@@ -472,15 +479,15 @@ class SrvImplDart implements Srv<SocketConnector> {
     // client side
     SocketConnector sc = await SocketConnector.serverToSocket(
       portA: localPort,
-      addressB: hosts[0],
+      addressB: relayAddress,
       portB: streamingPort,
-      verbose: false,
+      verbose: Platform.environment['SRV_TRACE'] == 'true',
       logger: ioSinkForLogger(logger),
       transformAtoB: encrypter,
       transformBtoA: decrypter,
       multi: multi,
       timeout: timeout,
-      beforeJoining: (Side sideA, Side sideB) async {
+      beforeJoining: (Side sideA, Side sideB) {
```

Comment:
has to be synchronous to avoid a race in `SocketConnector.handleSingleConnection`